### PR TITLE
Merge master -> stable

### PR DESCRIPTION
**Pull request by shrug**

### CODE REVIEW

**Comment by melissa** [comment]
I *think* this is good. The windows stuff looks okay. I'd like to check with people on core/windows before merging this though

**Comment by shrug** [comment]
Ping @Iristyle @joshcooper - anything look troubling in this merge?

**Comment by ScottGarman** [comment]
:+1:  all my huaweios stuff is in there.

File: README.md [hunk 1]
```diff
@@ -109,12 +101,12 @@ Note that for all merges from master or stable, the merge should pick up:
...
 
-Here's a sample snippet used for a stable -> aardwolf merge:
+Here's a sample snippet used for a stable -> master merge:
 
 ```
 git merge --no-commit --no-ff stable
-for i in {hiera,facter,puppet,marionette-collective,pxp-agent,cpp-pcp-client}; do git checkout aardwolf -- configs/components/$i.json;done
-git checkout aardwolf -- configs/components/windows_*.json
+for i in {hiera,facter,puppet,marionette-collective,pxp-agent,cpp-pcp-client}; do git checkout master -- configs/components/$i.json;done
+git checkout master -- configs/components/windows_puppet.json
```
Comment (joshcooper) [inline README.md]: why exclude `windows_ruby.json`?

File: bin/build-windows.rb [hunk 2]
```diff
@@ -78,7 +78,7 @@ def clone_and_rynsc_private_repo(fork, ref, hostname, ssh_key, component = nil)
 
 # Set up the environment so I don't keep crying
 ssh_command = "ssh #{ssh_key} -T -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null Administrator@#{hostname}"
-ssh_env = "export PATH=\'/cygdrive/c/Program Files/Git/cmd:/home/Administrator/deps/ruby-#{ruby_version}-#{ruby_arch}-mingw32/bin:/cygdrive/c/ProgramData/chocolatey/bin:/cygdrive/c/Program Files (x86)/Windows Installer XML v3.5/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/bin\'"
+ssh_env = "export PATH=\'/cygdrive/c/Program Files/Git/cmd:/home/Administrator/deps/ruby-#{ruby_version}-#{ruby_arch}-mingw32/bin:/cygdrive/c/ProgramData/chocolatey/bin:/cygdrive/c/Program Files (x86)/WiX Toolset v3.10/bin:/usr/local/bin:/usr/bin:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/bin\'"
```
Comment (joshcooper) [inline bin/build-windows.rb]: Does this mean the next 1.4.z puppet-agent release will be based off WiX 3.10? I'm thinking we don't want to do that until puppet-agent 1.5.0?

File: configs/components/puppet.rb [hunk 3]
```diff
@@ -1,6 +1,10 @@
 component "puppet" do |pkg, settings, platform|
   pkg.load_from_json("configs/components/puppet.json")
 
+  if platform.is_windows?
+    pkg.add_source("file://resources/files/windows/environment.bat", sum: "c698da37e559935e904f8690dd5c26fa")
```
Comment (joshcooper) [inline configs/components/puppet.rb]: what about adding environment.bat to puppet-agent itself, e.g. https://github.com/puppetlabs/puppet-agent/tree/master/resources/windows/msi/environment.bat

Comment (joshcooper) [inline configs/components/puppet.rb]: ha, nevermind!

File: configs/components/ruby.rb [hunk 4]
```diff
@@ -3,6 +3,11 @@
   pkg.md5sum "091b62f0a9796a3c55de2a228a0e6ef3"
   pkg.url "http://buildsources.delivery.puppetlabs.net/ruby-#{pkg.get_version}.tar.gz"
 
+  if platform.is_windows?
+    pkg.add_source "http://buildsources.delivery.puppetlabs.net/windows/elevate/elevate.exe", sum: "bd81807a5c13da32dd2a7157f66fa55d"
+    pkg.add_source "file://resources/files/windows/elevate.exe.config", sum: "a5aecf3f7335fa1250a0f691d754d561"
```
Comment (joshcooper) [inline configs/components/ruby.rb]: It's surprising to me that these are in two different places, but are related.

Comment (melissa) [inline configs/components/ruby.rb]: This is just to retrieve the files. The lines below are to install them

File: bin/build-windows.rb (hunk 2 above)
Comment (shrug) [inline bin/build-windows.rb]: If that's the case then we may want to hold off on the merge until then and cherry-pick the stuff we want (specifically the Huawei bits)?

Comment (shrug) [inline bin/build-windows.rb]: Or, actually, I think this won't impact any releases until we opt-in to the vanagon windows builds over puppet_for_the_win. @McdonaldSeanp @melissa ?

Comment (McdonaldSeanp) [inline bin/build-windows.rb]: /cc @Iristyle ^^^^^

**Pull request closed without merging**